engine/access/state_stream/backend/handler.go

```diff
@@ -213,3 +215,155 @@ func (h *Handler) GetRegisterValues(_ context.Context, request *executiondata.Ge
 
 	return &executiondata.GetRegisterValuesResponse{Values: values}, nil
 }
+
+// convertAccountsStatusesResultsToMessage converts account status responses to the message
+func convertAccountsStatusesResultsToMessage(
+	eventVersion entities.EventEncodingVersion,
+	resp *AccountStatusesResponse,
+) ([]*executiondata.SubscribeAccountStatusesResponse_Result, error) {
+	var results []*executiondata.SubscribeAccountStatusesResponse_Result
+	for address, events := range resp.AccountEvents {
+		convertedEvent, err := convert.EventsToMessagesWithEncodingConversion(events, entities.EventEncodingVersion_CCF_V0, eventVersion)
+		if err != nil {
+			return nil, status.Errorf(codes.Internal, "could not convert events to entity: %v", err)
+		}
+
+		results = append(results, &executiondata.SubscribeAccountStatusesResponse_Result{
+			Address: flow.HexToAddress(address).Bytes(),
+			Events:  convertedEvent,
+		})
+	}
+	return results, nil
+}
+
+// sendSubscribeAccountStatusesResponseFunc defines the function signature for sending account status responses
+type sendSubscribeAccountStatusesResponseFunc func(*executiondata.SubscribeAccountStatusesResponse) error
+
+// handleAccountStatusesResponse handles account status responses by converting them to the message and sending them to the subscriber.
+func (h *Handler) handleAccountStatusesResponse(
+	requestHeartbeatInterval uint64,
+	evenVersion entities.EventEncodingVersion,
+	send sendSubscribeAccountStatusesResponseFunc,
+) func(resp *AccountStatusesResponse) error {
+	heartbeatInterval := requestHeartbeatInterval
+	if heartbeatInterval == 0 {
+		heartbeatInterval = h.defaultHeartbeatInterval
+	}
+
+	blocksSinceLastMessage := uint64(0)
+
+	return func(resp *AccountStatusesResponse) error {
+		// check if there are any events in the response. if not, do not send a message unless the last
+		// response was more than HeartbeatInterval blocks ago
+		if len(resp.AccountEvents) == 0 {
+			blocksSinceLastMessage++
+			if blocksSinceLastMessage < heartbeatInterval {
+				return nil
+			}
+			blocksSinceLastMessage = 0
+		}
+
+		results, err := convertAccountsStatusesResultsToMessage(evenVersion, resp)
+		if err != nil {
+			return err
+		}
+
+		err = send(&executiondata.SubscribeAccountStatusesResponse{
+			BlockId:      convert.IdentifierToMessage(resp.BlockID),
+			BlockHeight:  resp.Height,
+			Results:      results,
+			MessageIndex: resp.MessageIndex,
```

Comment:
sorry, I'm just noticing now, but `MessageIndex` needs to be sequential for the responses sent to the client. Right now it's sequential for the messages returned by the backend, but some messages are dropped if there are not events. 

I think the logic for maintaining the message index needs to move into the handler.

The purpose of this field is provide a way for a client to detect if they missed a message, so it's important that the numbers are strictly sequential